lightning/src/ln/channelmanager.rs

```diff
@@ -4915,16 +4914,21 @@ where
 				let mut events = self.pending_events.lock().unwrap();
 				events.append(&mut new_intercept_events);
 			}
+			if push_forward_event { self.push_pending_forwards_ev() }
+		}
+	}
 
-			match forward_event {
-				Some(time) => {
-					let mut pending_events = self.pending_events.lock().unwrap();
-					pending_events.push(events::Event::PendingHTLCsForwardable {
-						time_forwardable: time
-					});
-				}
-				None => {},
-			}
+	// We only want to push a PendingHTLCsForwardable event if no others are queued.
+	fn push_pending_forwards_ev(&self) {
+		let mut pending_events = self.pending_events.lock().unwrap();
+		let forward_ev_exists = pending_events.iter()
```

Comment:
Ok I see the case we're not covering. I don't think we'd want `needs_abandon`, though, seems we'd want a new `needs_retry` check IIUC. Will add this. (Note that `needs_abandon` should only apply to restart since we set all payments as non-retryable on deser.)